### DISCUSSION TITLE
Minimum Requirements for Input Leap

### DISCUSSION THREAD

**Discussion by sithlord48** [input-leap/input-leap#1720] [Development of Input Leap]
I would like to use this as a place to talk about the plans for what minimum requirements we will have for input leap

My suggestions are:
  - C++17
  - CMake 3.21 
  - OpenSSL 3.1
  - Qt 6.2

Minimum OS Versions:
 - Windows 10
 - Linux (Debian 12)
 - macOS 11

Since this is a new fork let's take the opportunity to update some of our dependencies and work to drop some of the older systems. 

I think most Input-leap devs do not expect much in terms of the changes to the protocol and hope to be able to use Barrier with Input Leap if that is the case for the older systems we can recommend they use Barrier or a Flatpak install when the time comes.

(Edited by @shymega, tidied up grammar and lists)

**Comment by jkugler** [comment]
I would suggest: Whatever is not EOL. Debian 11 is still around and supported. Debian 10 is still in LTS. Maybe not RHEL/CentOS 7, as that goes EOL middle of next year. But Redhat 8/9 would certainly be in scope.

I know Qt 5.x is EOL; not sure how that affects the Open Source version.

**Comment by ph818** [comment]
Basically, input leap should be supported on older systems that are not yet EOL. 

If it really can be packaged in flatpak for older systems and still work, maybe it is ok to require newer dependencies.

I don't think it's a good idea to require new users install barrier, since it has significant...barriers to ease of use.
